java/ql/src/semmle/code/java/dataflow/FlowSources.qll

```diff
@@ -318,3 +319,34 @@ class AndroidIntentInput extends DataFlow::Node {
     )
   }
 }
+
+/** Method access to external inputs of `android.content.Intent` or `android.os.BaseBundle` object. */
+class IntentGetExtraMethodAccess extends MethodAccess {
+  IntentGetExtraMethodAccess() {
+    exists(AndroidComponent ac |
+      this.getEnclosingCallable().getDeclaringType() = ac and
+      ac.isExported() and
```

Comment:
Can we push this `isExported()` filter right into `AndroidIntentInput`, since that seems like the point where we know whether the source is dangerous or not

Reply:
Moved the the <code>isExported()</code> check to <code>AndroidIntentInput</code>.